barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_rollup_recursive_flavor.hpp

```diff
@@ -0,0 +1,144 @@
+            , ipa_claim_public_input_indices(native_key->ipa_claim_public_input_indices)
+        {
+            this->pcs_verification_key = native_key->pcs_verification_key;
+            this->circuit_size = native_key->circuit_size;
+            this->log_circuit_size = numeric::get_msb(this->circuit_size);
+            this->num_public_inputs = native_key->num_public_inputs;
+            this->pub_inputs_offset = native_key->pub_inputs_offset;
+            this->contains_pairing_point_accumulator = native_key->contains_pairing_point_accumulator;
+            this->pairing_point_accumulator_public_input_indices =
+                native_key->pairing_point_accumulator_public_input_indices;
+
+            // Generate stdlib commitments (biggroup) from the native counterparts
+            for (auto [commitment, native_commitment] : zip_view(this->get_all(), native_key->get_all())) {
+                commitment = Commitment::from_witness(builder, native_commitment);
+            }
+        };
+
+        /**
+         * @brief Deserialize a verification key from a vector of field elements
+         *
+         * @param builder
+         * @param elements
+         */
+        VerificationKey(CircuitBuilder& builder, std::span<FF> elements)
+        {
+            using namespace bb::stdlib::field_conversion;
+
+            size_t num_frs_read = 0;
+
+            this->circuit_size = uint64_t(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            this->num_public_inputs = uint64_t(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            this->pub_inputs_offset = uint64_t(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            this->contains_pairing_point_accumulator =
+                bool(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            for (uint32_t& idx : this->pairing_point_accumulator_public_input_indices) {
+                idx = uint32_t(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            }
+            contains_ipa_claim = bool(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            for (uint32_t& idx : this->ipa_claim_public_input_indices) {
+                idx = uint32_t(deserialize_from_frs<FF>(builder, elements, num_frs_read).get_value());
+            }
+
+            for (Commitment& commitment : this->get_all()) {
+                commitment = deserialize_from_frs<Commitment>(builder, elements, num_frs_read);
+            }
+        }
+
+        /**
+         * @brief Construct a VerificationKey from a set of corresponding witness indices
+         *
+         * @param builder
+         * @param witness_indices
+         * @return VerificationKey
+         */
+        static VerificationKey from_witness_indices(CircuitBuilder& builder,
+                                                    const std::span<const uint32_t>& witness_indices)
```

Comment:
We've gotten in trouble before passing references to spans--can you just pass by value?